### DISCUSSION TITLE
How to flexibly run Dag on a specified Worker node

### DISCUSSION THREAD

**Discussion by hrbust-liu** [apache/airflow#41947] [Q&A]
I have multiple workers and multiple dags. I expect that dag can parameterize the worker to be run (maybe configure a group of workers and only trigger the run in one of them). At the same time, even if multiple dags are submitted concurrently, even if their worker configurations conflict, a single worker only runs one dag at any time, and the rest of the dag tasks are queued.

**Comment by potiuk** [comment]
Look in airflow docs for "pools",  "queues" and for "slots". That should give you all te options. Generally speaking you can have workers belong to queues and have your tasks assigned to a specific queue. Also you can mark a task to take "n" slots in a pool and assign tasks to specific pools.

combination of those should give you all you need.

**Reply by hadarsharon** [reply]
@potiuk Not to hijack the thread but seeing as how the question is answered in a basic sense, could I perhaps provide a more elaborate follow-up question:

If we're using the Celery workers with HPA or KEDA, is there support in the Airflow Helm chart for scaling them on a per-queue basis? Kinda like worker groups for a given Airflow queue. 

I know that you can launch Celery workers that are assigned queues per se, but managing such deployment on Kubernetes with workers coming and going according to scale sounds more intricate, and I was wondering if that is somehow still within the scope of the Official Helm Chart (this could be of great use to me in my deployment).

**Reply by potiuk** [reply]
> If we're using the Celery workers with HPA or KEDA, is there support in the Airflow Helm chart for scaling them on a per-queue basis? Kinda like worker groups for a given Airflow queue.

No idea. Likely it would require some custom KEDA configuration and query.

**Reply by hrbust-liu** [reply]
I understand that queues and pools are predefined in Dag, rather than being controlled by parameters when tasks are executed.
However, my actual situation is this:
Suppose I have machines A, B, C, D, E, F, G. And I have Dags D1 and D2. There are multiple people triggering tasks, and each person may expect to trigger a Dag task on one of the machines in a group of machines. For example, trigger D1 on a trigger of machines ABCD, D1 in ADEF or D1 in G or D2 in ABF, etc. (number of combinations 2 * 2^7). At the same time, it is necessary to ensure that only one Dag task is running on the same machine at the same time.

**Reply by potiuk** [reply]
You cannot dynamically change Queue when you trigger the DAG, I am afraid - but theres is an issue for that and anyone is welcome to implement such a feature and contribute it https://github.com/apache/airflow/issues/35542
